backend/tests/unit/services/test_history_service.py

```
@@ -2334,4 +2334,341 @@ def test_get_session_conversation_history_no_interactions(self, history_service)
         assert session_conv is None
-        assert chat_conv is None
\ No newline at end of file
+        assert chat_conv is None
+    
+    # ===== Tests for Session History Formatting =====
+    
+    @pytest.mark.unit
+    def test_get_formatted_session_conversation_success(self, history_service):
+        """Test get_formatted_session_conversation returns formatted text."""
+        # Setup: Create session with LLM interaction
+        session = SessionFactory.create_test_session(
```

Ruff (0.14.14)

Local variable `session` is assigned to but never used

Remove assignment to unused variable `session`

(F841)